scripts/helmlint.sh

```diff
@@ -12,4 +12,18 @@ TARGET_DIR="${TEMP_DIR}/kubecf"
 mkdir "${TARGET_DIR}"
 
 tar xfz "${TARGET_FILE}" -C "${TEMP_DIR}"
-helm lint "${TARGET_DIR}"
+# Workaround: [Helm lint (v3) doesn't take values files into account during schema validation]
+# (https://github.com/helm/helm/issues/7756). Fixed in helm 3.3.0.
+perl -p -i -e 's/(system_domain:).*/$1 example.com/' "${TARGET_DIR}/values.yaml"
+helm lint "${TARGET_DIR}" --set system_domain=example.com
+
+# Running json schema validator a second time to also check the embedded config files.
+# Done in a separate run because the config files are merged here on top of values.yaml,
+# which will cause replacement of any arrays. Using `helm template` instead, which also
+# runs the validator, but doesn't suffer from bug #7756.
+
+# shellcheck disable=SC2046
+# We want word splitting with find.
+helm template "${TARGET_DIR}" --set system_domain=example.com \
+     --values "$(perl -e 'print join ",", @ARGV' -- $(find "${TARGET_DIR}/config" -type f))" \
+     > /dev/null
```

Comment:
Doesn't your `find` command miss a `/config` part on the starting directory?

Beyond that, I don't find your version more readable than mine, but that might be because I've written mine. 😄 

For example, the real command `helm template` is hidden in the middle of the second line, instead of being at the very front...

Reply:
> Doesn't your `find` command miss a `/config` part on the starting directory?

Err, yes, copy and paste error — I typed my command in a shell with the variable expanded, and hand-edited it incorrectly in the textbox.

The primary benefit of my version is dropping the extra shell expansion (which means dropping the shellcheck override).  That, and dropping perl, since the layers of string quoting is confusing :D  Not really all that much different though.

Reply:
I'm not convinced; I continue to find my version more readable, so don't plan to change it.

Try harder to convince me, or resolve...

Reply:
Look at it again, I find both versions kind of meh, so I'll just resolve.